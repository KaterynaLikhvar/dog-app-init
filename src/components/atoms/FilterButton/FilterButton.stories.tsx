import type { Meta, StoryObj } from '@storybook/react';
import { FilterButton } from '.';

const meta = {
  title: 'Atoms/FilterButton',
  component: FilterButton,
  argTypes: {
    state: {
      options: ['default', 'down', 'random']
    }
  }
} satisfies Meta<typeof FilterButton>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Up: Story = {
  args: {
    state: 'default'
  }
};

export const Down: Story = {
  args: {
    state: 'down'
  }
};

export const Random: Story = {
  args: {
    state: 'random'
  }
};
