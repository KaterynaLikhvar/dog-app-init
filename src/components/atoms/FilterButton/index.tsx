import { SvgIcon, styled } from '@mui/material';
import { FC } from 'react';
import theme from '../../../theme/theme';

import FilterIconUp from '../../../assets/FilterIconUp.svg?react';
import FilterIconDown from '../../../assets/FilterIconDown.svg?react';
import FilterIconRandom from '../../../assets/FilterIconRandom.svg?react';

const StyledFilterButton = styled('button')({
  backgroundColor: '#E6E6E6',
  padding: theme.spacing(1, 2),
  borderRadius: 6,
  boxShadow: '3px 3px 0px 0px #000;',
  border: '0.6px solid #090808',
  transition: 'box-shadow .2s ease',
  '&:hover': {
    boxShadow: '3px 3px 0px 0px #FFCF32;',
    border: '1px solid #FFCF32',
    backgroundColor: 'transparent',
    outline: 'none'
  },
  '&:active, &:focus': {
    boxShadow: 'none',
    border: '1px solid #FFCF32',
    backgroundColor: 'transparent',
    outline: '#FFCF32'
  }
});

type FilterButtonProps = {
  state: 'default' | 'down' | 'random';
};

export const FilterButton: FC<FilterButtonProps> = ({ state }) => {
  switch (state) {
    case 'down':
      return (
        <StyledFilterButton>
          <SvgIcon component={FilterIconDown} inheritViewBox />
        </StyledFilterButton>
      );
    case 'random':
      return (
        <StyledFilterButton>
          <SvgIcon component={FilterIconRandom} inheritViewBox />
        </StyledFilterButton>
      );
    default:
      return (
        <StyledFilterButton>
          <SvgIcon component={FilterIconUp} inheritViewBox />
        </StyledFilterButton>
      );
  }
};
