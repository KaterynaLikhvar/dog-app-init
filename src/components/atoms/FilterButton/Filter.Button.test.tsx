import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import * as stories from './FilterButton.stories';

describe('FilterButton', () => {
  const { Up, Down, Random } = composeStories(stories);
  it('should match snapshot Primary', () => {
    const { container } = render(<Up />);
    expect(container).toMatchSnapshot();
  });
  it('should match snapshot Secondary', () => {
    const { container } = render(<Down />);
    expect(container).toMatchSnapshot();
  });
  it('should match snapshot Secondary', () => {
    const { container } = render(<Random />);
    expect(container).toMatchSnapshot();
  });
});
