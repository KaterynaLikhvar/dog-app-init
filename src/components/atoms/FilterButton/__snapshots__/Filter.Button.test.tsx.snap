// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton should match snapshot Primary 1`] = `
.emotion-0 {
  background-color: #FBFAFC;
  padding: 8px 16px;
  border-radius: 6px;
  box-shadow: 3px 3px 0px 0px #000;
  border: 0.6px solid #090808;
}

.emotion-0:hover,
.emotion-0:active {
  box-shadow: 3px 3px 0px 0px #FFCF32;
  border: 1px solid #FFCF32;
  back-ground-color: none;
}

<div>
  <button
    class="emotion-0"
  >
    <svg
      fill="none"
      height="20"
      viewBox="0 0 18 20"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M11.6122 9.172C11.5322 9.172 11.4402 9.156 11.3363 9.124C11.2402 9.084 11.1602 9.032 11.0962 8.968C11.0402 8.904 11.0122 8.832 11.0122 8.752C11.0122 8.72 11.0162 8.688 11.0242 8.656L13.4842 0.676003C13.5162 0.556004 13.5922 0.468002 13.7122 0.412003C13.8322 0.356003 13.9602 0.328003 14.0962 0.328003C14.2402 0.328003 14.3722 0.360003 14.4922 0.424004C14.6122 0.480003 14.6882 0.564003 14.7202 0.676003L17.1682 8.656C17.1842 8.688 17.1922 8.716 17.1922 8.74C17.1922 8.82 17.1562 8.892 17.0842 8.956C17.0202 9.02 16.9402 9.072 16.8442 9.112C16.7482 9.152 16.6562 9.172 16.5682 9.172C16.4802 9.172 16.4042 9.156 16.3403 9.124C16.2762 9.084 16.2323 9.024 16.2082 8.944L15.6682 7.156H12.5122L11.9962 8.944C11.9722 9.024 11.9242 9.084 11.8522 9.124C11.7882 9.156 11.7082 9.172 11.6122 9.172ZM15.4402 6.328L14.0962 1.816L12.7522 6.328H15.4402Z"
        fill="#5F556D"
        fill-rule="evenodd"
      />
      <path
        d="M3.80824 0.672003C4.01512 0.672003 4.20063 0.799417 4.27491 0.992514L6.77491 7.49251C6.87403 7.75025 6.74546 8.03955 6.48774 8.13868C6.22999 8.23781 5.94068 8.10923 5.84156 7.85149L4.30824 3.86484V19.172C4.30824 19.4481 4.08439 19.672 3.80824 19.672C3.53209 19.672 3.30824 19.4481 3.30824 19.172V3.86484L1.77491 7.85149C1.67579 8.10923 1.38649 8.23781 1.12873 8.13868C0.871013 8.03955 0.742443 7.75025 0.841564 7.49251L3.34156 0.992514C3.41584 0.799417 3.60136 0.672003 3.80824 0.672003Z"
        fill="#5F556D"
      />
      <path
        clip-rule="evenodd"
        d="M11.7802 19.076C11.8682 19.14 11.9802 19.172 12.1163 19.172H14.4562C14.9042 19.172 15.3082 19.092 15.6682 18.932C16.0282 18.772 16.3122 18.512 16.5202 18.152C16.7283 17.784 16.8322 17.292 16.8322 16.676V16.46C16.8322 16.044 16.7802 15.708 16.6763 15.452C16.5722 15.188 16.4282 14.976 16.2442 14.816C16.0603 14.648 15.8442 14.516 15.5962 14.42C15.9082 14.284 16.1562 14.048 16.3403 13.712C16.5242 13.376 16.6163 12.956 16.6163 12.452C16.6163 11.892 16.5043 11.464 16.2802 11.168C16.0643 10.872 15.7682 10.672 15.3922 10.568C15.0242 10.456 14.6203 10.4 14.1802 10.4H12.1163C12.0043 10.4 11.9002 10.432 11.8042 10.496C11.7082 10.552 11.6602 10.64 11.6602 10.76V18.824C11.6602 18.92 11.7002 19.004 11.7802 19.076ZM14.2882 18.26H12.6682V14.84H14.2882C14.7762 14.84 15.1522 14.96 15.4162 15.2C15.6882 15.432 15.8242 15.836 15.8242 16.412V16.616C15.8242 17.184 15.6882 17.6 15.4162 17.864C15.1522 18.128 14.7762 18.26 14.2882 18.26ZM14.1682 14H12.6682V11.312H14.1322C14.4282 11.312 14.6842 11.352 14.9002 11.432C15.1242 11.504 15.2962 11.636 15.4162 11.828C15.5442 12.012 15.6082 12.268 15.6082 12.596C15.6082 13.092 15.4922 13.452 15.2602 13.676C15.0282 13.892 14.6642 14 14.1682 14Z"
        fill="#5F556D"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`FilterButton should match snapshot Secondary 1`] = `
<div>
  <svg
    fill="none"
    height="39"
    viewBox="0 0 61 39"
    width="61"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      filter="url(#filter0_d_16_980)"
    >
      <rect
        fill="#E6E6E6"
        height="36"
        rx="6"
        shape-rendering="crispEdges"
        width="58"
      />
      <path
        clip-rule="evenodd"
        d="M31.6122 16.172C31.5322 16.172 31.4402 16.156 31.3362 16.124C31.2402 16.084 31.1602 16.032 31.0962 15.968C31.0402 15.904 31.0122 15.832 31.0122 15.752C31.0122 15.72 31.0162 15.688 31.0242 15.656L33.4842 7.676C33.5162 7.556 33.5922 7.468 33.7122 7.412C33.8322 7.356 33.9602 7.328 34.0962 7.328C34.2402 7.328 34.3722 7.36 34.4922 7.424C34.6122 7.48 34.6882 7.564 34.7202 7.676L37.1682 15.656C37.1842 15.688 37.1922 15.716 37.1922 15.74C37.1922 15.82 37.1562 15.892 37.0842 15.956C37.0202 16.02 36.9402 16.072 36.8442 16.112C36.7482 16.152 36.6562 16.172 36.5682 16.172C36.4802 16.172 36.4042 16.156 36.3402 16.124C36.2762 16.084 36.2322 16.024 36.2082 15.944L35.6682 14.156H32.5122L31.9962 15.944C31.9722 16.024 31.9242 16.084 31.8522 16.124C31.7882 16.156 31.7082 16.172 31.6122 16.172ZM35.4402 13.328L34.0962 8.816L32.7522 13.328H35.4402Z"
        fill="#5F556D"
        fill-rule="evenodd"
      />
      <path
        d="M23.8082 26.672C23.6014 26.672 23.4158 26.5446 23.3416 26.3515L20.8416 19.8515C20.7424 19.5938 20.871 19.3045 21.1288 19.2053C21.3865 19.1062 21.6758 19.2348 21.7749 19.4925L23.3082 23.4792V8.172C23.3082 7.89586 23.5321 7.672 23.8082 7.672C24.0844 7.672 24.3082 7.89586 24.3082 8.172V23.4792L25.8416 19.4925C25.9407 19.2348 26.23 19.1062 26.4877 19.2053C26.7455 19.3045 26.874 19.5938 26.7749 19.8515L24.2749 26.3515C24.2006 26.5446 24.0151 26.672 23.8082 26.672Z"
        fill="#5F556D"
      />
      <path
        clip-rule="evenodd"
        d="M31.7802 26.076C31.8682 26.14 31.9802 26.172 32.1162 26.172H34.4562C34.9042 26.172 35.3082 26.092 35.6682 25.932C36.0282 25.772 36.3122 25.512 36.5202 25.152C36.7282 24.784 36.8322 24.292 36.8322 23.676V23.46C36.8322 23.044 36.7802 22.708 36.6762 22.452C36.5722 22.188 36.4282 21.976 36.2442 21.816C36.0602 21.648 35.8442 21.516 35.5962 21.42C35.9082 21.284 36.1562 21.048 36.3402 20.712C36.5242 20.376 36.6162 19.956 36.6162 19.452C36.6162 18.892 36.5042 18.464 36.2802 18.168C36.0642 17.872 35.7682 17.672 35.3922 17.568C35.0242 17.456 34.6202 17.4 34.1802 17.4H32.1162C32.0042 17.4 31.9002 17.432 31.8042 17.496C31.7082 17.552 31.6602 17.64 31.6602 17.76V25.824C31.6602 25.92 31.7002 26.004 31.7802 26.076ZM34.2882 25.26H32.6682V21.84H34.2882C34.7762 21.84 35.1522 21.96 35.4162 22.2C35.6882 22.432 35.8242 22.836 35.8242 23.412V23.616C35.8242 24.184 35.6882 24.6 35.4162 24.864C35.1522 25.128 34.7762 25.26 34.2882 25.26ZM34.1682 21H32.6682V18.312H34.1322C34.4282 18.312 34.6842 18.352 34.9002 18.432C35.1242 18.504 35.2962 18.636 35.4162 18.828C35.5442 19.012 35.6082 19.268 35.6082 19.596C35.6082 20.092 35.4922 20.452 35.2602 20.676C35.0282 20.892 34.6642 21 34.1682 21Z"
        fill="#5F556D"
        fill-rule="evenodd"
      />
      <rect
        height="35.4"
        rx="5.7"
        shape-rendering="crispEdges"
        stroke="#090808"
        stroke-width="0.6"
        width="57.4"
        x="0.3"
        y="0.3"
      />
    </g>
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="userSpaceOnUse"
        height="39"
        id="filter0_d_16_980"
        width="61"
        x="0"
        y="0"
      >
        <feflood
          flood-opacity="0"
          result="BackgroundImageFix"
        />
        <fecolormatrix
          in="SourceAlpha"
          result="hardAlpha"
          type="matrix"
          values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
        />
        <feoffset
          dx="3"
          dy="3"
        />
        <fecomposite
          in2="hardAlpha"
          operator="out"
        />
        <fecolormatrix
          type="matrix"
          values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0"
        />
        <feblend
          in2="BackgroundImageFix"
          mode="normal"
          result="effect1_dropShadow_16_980"
        />
        <feblend
          in="SourceGraphic"
          in2="effect1_dropShadow_16_980"
          mode="normal"
          result="shape"
        />
      </filter>
    </defs>
  </svg>
</div>
`;

exports[`FilterButton should match snapshot Secondary 2`] = `
<div>
  <svg
    fill="none"
    height="39"
    viewBox="0 0 61 39"
    width="61"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      filter="url(#filter0_d_16_986)"
    >
      <rect
        fill="#E6E6E6"
        height="36"
        rx="6"
        shape-rendering="crispEdges"
        width="58"
      />
      <path
        d="M20.8259 10.0326C20.9994 9.91989 21.2243 9.92571 21.3918 10.0472L27.0286 14.137C27.2521 14.2991 27.3019 14.6118 27.1397 14.8353C26.9775 15.0588 26.6649 15.1085 26.4414 14.9464L22.9841 12.438L31.321 25.2757C31.4714 25.5072 31.4056 25.8169 31.174 25.9673C30.9424 26.1177 30.6327 26.0519 30.4823 25.8203L22.1455 12.9826L23.0308 17.1612C23.088 17.4314 22.9154 17.6968 22.6453 17.754C22.3751 17.8112 22.1097 17.6386 22.0525 17.3685L20.609 10.5555C20.5661 10.3531 20.6523 10.1452 20.8259 10.0326Z"
        fill="#5F556D"
      />
      <path
        d="M37.174 10.0326C37.0005 9.91989 36.7755 9.92571 36.608 10.0472L30.9712 14.137C30.7477 14.2991 30.698 14.6118 30.8601 14.8353C31.0223 15.0588 31.3349 15.1085 31.5585 14.9464L35.0157 12.438L26.6788 25.2757C26.5284 25.5072 26.5942 25.8169 26.8258 25.9673C27.0574 26.1177 27.3671 26.0519 27.5175 25.8203L35.8544 12.9826L34.969 17.1612C34.9118 17.4314 35.0844 17.6968 35.3545 17.754C35.6247 17.8112 35.8901 17.6386 35.9473 17.3685L37.3908 10.5555C37.4337 10.3531 37.3475 10.1452 37.174 10.0326Z"
        fill="#5F556D"
      />
      <rect
        height="35.4"
        rx="5.7"
        shape-rendering="crispEdges"
        stroke="#090808"
        stroke-width="0.6"
        width="57.4"
        x="0.3"
        y="0.3"
      />
    </g>
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="userSpaceOnUse"
        height="39"
        id="filter0_d_16_986"
        width="61"
        x="0"
        y="0"
      >
        <feflood
          flood-opacity="0"
          result="BackgroundImageFix"
        />
        <fecolormatrix
          in="SourceAlpha"
          result="hardAlpha"
          type="matrix"
          values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
        />
        <feoffset
          dx="3"
          dy="3"
        />
        <fecomposite
          in2="hardAlpha"
          operator="out"
        />
        <fecolormatrix
          type="matrix"
          values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0"
        />
        <feblend
          in2="BackgroundImageFix"
          mode="normal"
          result="effect1_dropShadow_16_986"
        />
        <feblend
          in="SourceGraphic"
          in2="effect1_dropShadow_16_986"
          mode="normal"
          result="shape"
        />
      </filter>
    </defs>
  </svg>
</div>
`;
